### PR TITLE
Hbo presto on spark integration

### PR DESCRIPTION
**Pull request by pranjalssh**
Depends on https://github.com/prestodb/presto/pull/18584

Integrate HBO framework with presto on spark

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by highker** [comment]
lol, don't call it sapphire. It is presto on spark

File: presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkPageOutputOperator.java [hunk 1]
```diff
@@ -185,6 +185,7 @@ public void addInput(Page page)
                 .collect(toImmutableList());
 
         serializedPages.forEach(outputBuffer::enqueue);
+        operatorContext.recordOutput(page.getSizeInBytes(), page.getPositionCount());
```
Comment (highker) [inline presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkPageOutputOperator.java:188]: Are these existing bugs or something? @pgupta2 could you help to verify?

**Review by highker — COMMENTED** [review]
Please also fix the "Sapphire" word in the commit title.

LGTM. Not approving for now in case I accidentally merge the first commit.

File: presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkPageOutputOperator.java (hunk 1 above)
Comment (pgupta2) [reply presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkPageOutputOperator.java:188]: This is going to record the output bytes/rows for data that is being shuffled, which is currently missing from operator stats in PoS. Nice catch!!

**Review by pgupta2 — APPROVED** [review]
LGTM.

**Review by rschlussel — APPROVED** [review]

**Pull request merged**